### DISCUSSION TITLE
Practice to run the Git repos Arena + Minestom

### DISCUSSION THREAD

**Discussion by cristian-spiescu** [Minestom/Minestom#1391] [General]
Hi there!

This is my first post here, so let me express my appreciation towards the team.
I'm starting to learn Minestom. I discovered this project at the beginning of 2022, and I found it really interesting. Now, about 8 month later I can say that I'm deeply impressed by 1) the progress [diff :)] that happened meanwhile and 2) by the quality of the code. W/ my 18 years of Java experience, I can say that I rarely seen a Java project so nice to read. 

I am new to Gradle. I checked out the 2 Git repos, one for Arena, one for Minestom. I'm using Eclipse. In order to start Arena that uses the checked out Minestom (and not `Minestom.jar`), I used the [composite builds](https://docs.gradle.org/current/userguide/composite_builds.html) concept in the following way:

![image](https://user-images.githubusercontent.com/4469897/189297664-b268f778-42ee-4825-b04e-6acdc7fc0a9b.png)

I achieved my goal. But I'm wondering if this is the correct practice to employ. What I don't like, is that I had to modify the files in the Arena repo. Both for changing the artifact name + the "includeBuild" statement.

Thanks in advance,
Cristian.

**Comment by cristian-spiescu** [comment]
Hello,

Does anyone has an opinion about this? I guess that running from the git repos: arena + minestom is a common use case for a lot of people.

Regards,
Cristian

**Reply by FluxCapacitor2** [reply]
To me, this seems like the best way to do it. The only alternative way I can think of is to go into the Minestom project and run `./gradlew publishToMavenLocal` every time you update the Minestom code, and then update Arena's build file to use `net.minestom.server` version `1.0`.

**Comment by cristian-spiescu** [comment]
Thanks for your reply! 
An regarding the discrepancy between the artifact name `net.minestom.server` and `com.github.Minestom`: this comes from Jitpack, which for the free version doesn't allow a custom domain name, right?

**Reply by FluxCapacitor2** [reply]
Using `./gradlew publishToMavenLocal` will build the project and place it in your *local* repository, which means you can always find the resource without an internet connection and it's not downloaded from anywhere. This means you can specify whatever group ID you want, and Minestom gives a group ID of `net.minestom.server` as seen here:
https://github.com/Minestom/Minestom/blob/18c46481f4b7915e18ee005c9e57f0c8073f3ba6/build.gradle.kts#L7-L11
Jitpack overrides the group ID, setting it to `com.github.Minestom`, and it sets the version number to the latest Git commit hash. However, when you build and publish it locally, you are not using Jitpack, so these changes do not apply.

If you want to use your Minestom fork in a place that's not on the computer where it was compiled, the easiest option is to fork Minestom on GitHub and then use JitPack to import it.

**Comment by cristian-spiescu** [comment]
Thanks a lot!
